app/frontend/src/components/designer/FormDesigner.vue

```diff
@@ -382,6 +399,17 @@ export default {
         this.getFormSchema();
       }
     },
+    beforeDestroy() {
+      window.onbeforeunload = null;
+    }
+  },
+  watch: {
+    designerStep(newValue) {
```

Comment:
On Safari and Chrome, this isn't kicking in for me.
Going to the submitter side, I get the browser warning when leaving, but can't get it going in designer.

Reply:
This is a vue-router navigation issue. Will be looked at in future PR.